### PR TITLE
General code improvement

### PR DESCRIPTION
**Pull request by pzhanggit**

### CODE REVIEW

**Comment by allaffa** [comment]
@pzhanggit 

I am using the following configuration file:

{
    "Dataset": {
        "name": "CuAu_32atoms",
        "format": "LSMS",
        "num_nodes": 32,
        "node_features": {
            "name": ["num_of_protons","charge_density", "magnetic_moment"],
            "dim": [1,1,1],
            "column_index": [0,5,6]
        },
        "graph_features":{
            "name": [ "free_energy"],
            "dim": [1],
            "column_index": [0]
        }
    },
    "NeuralNetwork": {
        "Architecture": {
            "model_type": "PNN",
            "radius": 7,
            "max_neighbours": 5,
            "hidden_dim": 5,
            "num_conv_layers": 6,
            "num_sharedlayers": 2,
            "task_weights": [1.0]
        },
        "Variables_of_interest": {
            "input_node_features": [0],
            "output_names": ["charge_density"],
            "output_index": [1],
            "type": ["node"],
            "denormalize_output": "True"
        },
        "Training": {
            "num_epoch": 20,
            "perc_train": 0.7,
            "learning_rate": 1e-3,
            "batch_size": 64,
            "continue": 0,
            "startfrom": "existing_model"
        }
    }
}

So I am using GCNN to predict only the charge density. This is the error I get at the end of the training. 

Epoch: 19, Train MAE: 0.37515516, Val MAE: 0.08111136, Test RMSE: 0.08111136
100%|███████████████████████████████████████████| 75/75 [00:01<00:00, 54.41it/s]
Length of true and predicted values array is not the same!
DDP has to be initialized within a job - Running in sequential mode

**Comment by pzhanggit** [comment]
> @pzhanggit
> 
> I am using the following configuration file:
> 
> {
> "Dataset": {
> "name": "CuAu_32atoms",
> "format": "LSMS",
> "num_nodes": 32,
> "node_features": {
> "name": ["num_of_protons","charge_density", "magnetic_moment"],
> "dim": [1,1,1],
> "column_index": [0,5,6]
> },
> "graph_features":{
> "name": [ "free_energy"],
> "dim": [1],
> "column_index": [0]
> }
> },
> "NeuralNetwork": {
> "Architecture": {
> "model_type": "PNN",
> "radius": 7,
> "max_neighbours": 5,
> "hidden_dim": 5,
> "num_conv_layers": 6,
> "num_sharedlayers": 2,
> "task_weights": [1.0]
> },
> "Variables_of_interest": {
> "input_node_features": [0],
> "output_names": ["charge_density"],
> "output_index": [1],
> "type": ["node"],
> "denormalize_output": "True"
> },
> "Training": {
> "num_epoch": 20,
> "perc_train": 0.7,
> "learning_rate": 1e-3,
> "batch_size": 64,
> "continue": 0,
> "startfrom": "existing_model"
> }
> }
> }
> 
> So I am using GCNN to predict only the charge density. This is the error I get at the end of the training.
> 
> Epoch: 19, Train MAE: 0.37515516, Val MAE: 0.08111136, Test RMSE: 0.08111136
> 100%|███████████████████████████████████████████| 75/75 [00:01<00:00, 54.41it/s]
> Length of true and predicted values array is not the same!
> DDP has to be initialized within a job - Running in sequential mode

@allaffa  Fixed in 13abec

**Review by allaffa — APPROVED** [review]

**Comment by allaffa** [comment]
@pzhanggit 

Thank you. I ran the code using up to 10 GPUs on the new DGX machine up in SNS. Everything seems to be working fine. 
I am going to approve the merge request.
